### PR TITLE
Converted compression to cython and added linux/windows + python 3.7/3.8 buildspecs

### PR DESCRIPTION
**Pull request by tannermpeterson**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,12 +1,50 @@
...
+else:
+    USE_CYTHON = True
+
+# cmdclass = {}
+# ext_modules = []
+
+# if use_cython:
+#     ext_modules += [
+#         Extension("mypackage.mycythonmodule", ["cython/mycythonmodule.pyx"]),
+#     ]
```
Comment (ejfine) [inline setup.py]: clean up unnecessary commented code

File: setup.py [hunk 2]
```diff
@@ -1,12 +1,50 @@
...
+#     cmdclass.update({'build_ext': build_ext})
+# else:
+#     ext_modules += [
+#         Extension("mypackage.mycythonmodule", ["cython/mycythonmodule.c"]),
+#     ]
+
+
+ext = ".pyx" if USE_CYTHON else ".c"
+# os.path.join("src", "mantarray_waveform_analysis", "compression_cy.pyx")
```
Comment (ejfine) [inline setup.py]: ditto on the commented code

File: setup.py [hunk 3]
```diff
@@ -32,4 +70,11 @@
         "Programming Language :: Python :: 3.8",
         "Topic :: Scientific/Engineering",
     ],
+    ext_modules=extensions
+    # cythonize(
```
Comment (ejfine) [inline setup.py]: we don't want to just have this as commented code.  but we do need to keep the notes somewhere about the python dev module and the annotate=True...as those are helpful 'gotchas' that we'll avoid having to look up as we do more Cython

File: src/mantarray_waveform_analysis/__init__.py [hunk 4]
```diff
@@ -25,6 +24,14 @@
 from .transforms import FILTER_CHARACTERISTICS
 from .transforms import noise_cancellation
 
+
+if (
+    6 < 9
+):  # pragma: no cover # protect this from zimports deleting the pylint disable statement
+    from .compression_cy import (  # pylint: disable=import-error # Eli (8/18/20) unsure why pylint is unable to recognize cython import...
```
Comment (ejfine) [inline src/mantarray_waveform_analysis/__init__.py:30]: did you have a chance at all to take a look at why pylint can't recognize the cython import?

File: src/mantarray_waveform_analysis/compression_cy.pyx [hunk 5]
```diff
@@ -7,8 +9,8 @@
 
 R_SQUARE_CUTOFF = 0.95
 
-
-def rsquared(x_values: NDArray[int], y_values: NDArray[int]) -> float:
+#def rsquared(x_values: NDArray[int], y_values: NDArray[int]) -> float:
```
Comment (ejfine) [inline src/mantarray_waveform_analysis/compression_cy.pyx]: delete commented code

File: src/mantarray_waveform_analysis/compression_cy.pyx [hunk 6]
```diff
@@ -21,13 +23,41 @@ def rsquared(x_values: NDArray[int], y_values: NDArray[int]) -> float:
...
+    ss_tot = 0
+    cdef int y_sum,num_values,i
+    y_sum=0
+    num_values=y_values.shape[0]
+    ss_res=0
+    for i in range(num_values):
+        y_sum += y_values[i]
+        ss_res += (x_values[i] * slope + intercept - y_values[i]) ** 2
+    y_bar = y_sum / num_values
+    #y_bar = np.mean(y_values)
```
Comment (ejfine) [inline src/mantarray_waveform_analysis/compression_cy.pyx]: delete commented code

File: src/mantarray_waveform_analysis/compression_cy.pyx [hunk 7]
```diff
@@ -21,13 +23,41 @@ def rsquared(x_values: NDArray[int], y_values: NDArray[int]) -> float:
...
+
+    #ss_res = np.sum(((x_values * slope + intercept) - y_values) ** 2)
+    ss_tot = 0
+    for i in range(num_values):
+        ss_tot += (y_values[i] - y_bar) ** 2
+    #ss_tot = np.sum((y_values - y_bar) ** 2)
+
+    return 1 - ss_res / ss_tot
```
Comment (tannermpeterson) [inline src/mantarray_waveform_analysis/compression_cy.pyx:52]: @eli88fine can we delete everything after this first return statement?

**Review by ejfine — APPROVED** [review]

**Pull request merged**